lib/openstack_query/docs/QUERIES.md

```diff
@@ -0,0 +1,232 @@
+# Queries
+
+The class diagram for how a Query object is created is shown below (see [API.md](API.md) for usage)
+
+![Diagram Showing how Query Object is created](./imgs/queryapi-workflow.png)
+
+
+# Mappings
+
+Each Query has a corresponding Query Mapping which is responsible for defining its behaviour.
+These mappings include:
+ - what property(ies) are valid to query for/with
+   - and how to get those values from the equivalent Openstack object
+ - what preset-property pairs are valid for `where()` calls.
+ - what shared common properties (if any) are available between this resource and other resources - for chaining
+ - how to actually run the query using openstacksdk
+
+Each Query has a corresponding file called `<name_of_resource>_mapping`. The files is located in `/lib/openstack_query/mappings/`
+This file is used to configure the above.
+
+**Note**: see [PROPERTIES.md](PROPERTIES.md) for how to change property mappings
+
+## Reference
+
+#### get\_chain\_mappings
+
+This method allows us to define how to chain between these two different queries.
+See [CHAINING.md](CHAINING.md) for details
+
+#
+### get\_runner\_mapping
+
+This method allows us to attach a `Runner` Class which will be used to run the command against
+openstacksdk and handle any meta-parameters.
+See [RUNNERS.md](RUNNERS.md) for details
+
+#
+### get\_prop\_mapping
+
+This method allows us to attach a `PropEnum` Class which will be used to specify which properties the query
+can work on and handle any mappings.
+See [PROPERTIES.md](PROPERTIES.md) for details
+
+#
+#### get\_server\_side\_handler
+
+This method allows us to configure server-side key-word arguments to pass directly to Openstack API servers so that they
+can execute the query at the server-side. This allows for much quicker results than using client-side filter functions.
+See [PRESETS.md](PRESETS.md) for details
+
+
+#
+#### get_client_side_handler
+
+This method is used to define valid preset-property pairs that can be passed to `where()`.
+See PRESETS.md for details
```

Comment:
These are abstract methods to enforce their implementation - let the ABC force future devs to do the right thing

If these descriptions are better than the existing doc strings we should move them there into the code (which you'll be directly interacting with) rather than here

I think saying, implement the abstract class X, then letting the user read the interface is acceptable and keeps the docstrings near the codebase to save the person switching between this page and the new class they're creating